projects/self-creation-portal/src/app/components/resource-holder/resource-holder.component.ts

```diff
@@ -1,27 +1,39 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { MatButtonModule } from '@angular/material/button';
 import { MatCardModule } from '@angular/material/card';
 import { MatIconModule } from '@angular/material/icon';
 import { MatListModule } from '@angular/material/list';
 import { MatSidenavModule } from '@angular/material/sidenav';
 import { MatToolbarModule } from '@angular/material/toolbar';
-import { HeaderComponent, SideNavbarComponent } from 'lib-shared-modules';
+import { HeaderComponent, SideNavbarComponent, CardComponent, SearchComponent, PaginationComponent, FilterComponent } from '../../../../../lib-shared-modules/src/public-api';
+import { FormControl } from '@angular/forms';
+
 
 @Component({
   selector: 'app-resource-holder',
   standalone: true,
-  imports: [HeaderComponent,SideNavbarComponent,MatSidenavModule, MatButtonModule, MatIconModule, MatToolbarModule, MatListModule, MatCardModule],
+  imports: [HeaderComponent,SideNavbarComponent, CardComponent, SearchComponent, PaginationComponent, FilterComponent, MatSidenavModule, MatButtonModule, MatIconModule, MatToolbarModule, MatListModule, MatCardModule],
   templateUrl: './resource-holder.component.html',
   styleUrl: './resource-holder.component.scss',
 })
-export class ResourceHolderComponent {
+export class ResourceHolderComponent implements OnInit{
   backButton : boolean = true;
   subHeader : any;
   headerData : any = {};
   selctedCardItem : any;
   titleObj = {
     "title" : "Creation Portal"
   }
+  totalCount = 0;
+  pageSize = 5;
+  pageSizeOptions = [5, 10, 20, 100];
+  currentPage = 0;
+  currentSearch = '';
+  filteredLists: any[] = [];
+  currentFilters: any = {};
+  searchControl = new FormControl();
+  currentList: any[] = [];
+  setPaginatorToFirstpage: boolean = false;
```

Comment:
do we need this property ? can't we get status by current page itself ?

Reply:
I am using this for pagination issue, so it is required to handle the property binding.

Reply:
i didn't get so we get over a call and sort it out.